Lib/javascript/v8/javascriptrun.swg

```diff
@@ -289,8 +289,12 @@ SWIGRUNTIME int SWIG_V8_ConvertInstancePtr(SWIGV8_OBJECT objRef, void **ptr, swi
     if(!type_valid) {
       return SWIG_TypeError;
     }
+    int newmemory = 0;
+    *ptr = SWIG_TypeCast(tc, cdata->swigCObject, &newmemory);
```

Comment:
What jumps at me is that it makes you wonder if `newmemory` would require handling. If I "swig" your snippet and look, I see that conversion functions don't use `newmemory` in any way. It's probably used in other languages, but it's irrelevant in the context. And given these specific javascript circumstances I for one would argue that passing `NULL` instead of `&newmemory` would be a slightly better approach. Because **if** something changes and conversion subroutines would **start** using `newmemory`, it will be denoted by SEGV, which will inevitably prompt maintainers to look at this exact spot and figure out proper way to handle it. I mean if it doesn't SEGV, and `newmemory` would actually require some handling, then the problem might slip through the cracks and remain unnoticed for a longer period of time.

Disclaimer. I'm not swig maintainer, and the above comment might be misaligned with maintainers' opinions. But if the suggested approach is found sensible and applied, it would also be appropriate to leave a short "why-NULL" note in commentary.

Reply:
When #1365 is incorporated, the newmemory flag will be needed. It is used by the smart pointer casts, where you can't simply cast one pointer to another as the objects are actually pointers to smart pointers where casting requires creating a new smart pointer containing the cast pointer. Not sure if that makes sense, but take a look at say some Python wrappers wrapping shared_ptr.

Without the smart pointer support, I agree NULL would be better because this patch does not handle the case of newmemory being set.

Reply:
The shared pointer patch handles the newmemory variable, we may just reuse it here:

https://github.com/swig/swig/pull/1365/files#diff-339ae135626830816428ae1daef3325fa3eb5ab3ebb9a3aba2c10c22fcad3dd9R138